{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4896bf41-f8ba-4c72-a282-f90d14d873f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import os\n",
    "import azure.cosmos.documents as documents\n",
    "import azure.cosmos.cosmos_client as cosmos_client\n",
    "import azure.cosmos.exceptions as exceptions\n",
    "from azure.cosmos.partition_key import PartitionKey\n",
    "import datetime, requests\n",
    "import json\n",
    "from getpass import getpass\n",
    "import subprocess\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5a982503-9632-4b5c-b132-fc160baf384e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdin",
     "output_type": "stream",
     "text": [
      "Digite la ip:  45\n",
      "Digite el nombre del contenedor:  dfe\n"
     ]
    },
    {
     "ename": "ConnectionError",
     "evalue": "HTTPConnectionPool(host='127.0.0.1', port=5000): Max retries exceeded with url: /bluekeep_Dectect_Udec (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f89c8fc2790>: Failed to establish a new connection: [Errno 111] Connection refused'))",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mConnectionRefusedError\u001b[0m                    Traceback (most recent call last)",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/urllib3/connection.py:159\u001b[0m, in \u001b[0;36mHTTPConnection._new_conn\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    158\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 159\u001b[0m     conn \u001b[38;5;241m=\u001b[39m \u001b[43mconnection\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcreate_connection\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    160\u001b[0m \u001b[43m        \u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_dns_host\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mport\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mextra_kw\u001b[49m\n\u001b[1;32m    161\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    163\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m SocketTimeout:\n",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/urllib3/util/connection.py:84\u001b[0m, in \u001b[0;36mcreate_connection\u001b[0;34m(address, timeout, source_address, socket_options)\u001b[0m\n\u001b[1;32m     83\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m err \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m---> 84\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m err\n\u001b[1;32m     86\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m socket\u001b[38;5;241m.\u001b[39merror(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mgetaddrinfo returns an empty list\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/urllib3/util/connection.py:74\u001b[0m, in \u001b[0;36mcreate_connection\u001b[0;34m(address, timeout, source_address, socket_options)\u001b[0m\n\u001b[1;32m     73\u001b[0m     sock\u001b[38;5;241m.\u001b[39mbind(source_address)\n\u001b[0;32m---> 74\u001b[0m \u001b[43msock\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect\u001b[49m\u001b[43m(\u001b[49m\u001b[43msa\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     75\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m sock\n",
      "\u001b[0;31mConnectionRefusedError\u001b[0m: [Errno 111] Connection refused",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mNewConnectionError\u001b[0m                        Traceback (most recent call last)",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/urllib3/connectionpool.py:670\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    669\u001b[0m \u001b[38;5;66;03m# Make the request on the httplib connection object.\u001b[39;00m\n\u001b[0;32m--> 670\u001b[0m httplib_response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_request\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    671\u001b[0m \u001b[43m    \u001b[49m\u001b[43mconn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    672\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    673\u001b[0m \u001b[43m    \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    674\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout_obj\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    675\u001b[0m \u001b[43m    \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    676\u001b[0m \u001b[43m    \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    677\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    678\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    680\u001b[0m \u001b[38;5;66;03m# If we're going to release the connection in ``finally:``, then\u001b[39;00m\n\u001b[1;32m    681\u001b[0m \u001b[38;5;66;03m# the response doesn't need to know about the connection. Otherwise\u001b[39;00m\n\u001b[1;32m    682\u001b[0m \u001b[38;5;66;03m# it will also try to release it and we'll have a double-release\u001b[39;00m\n\u001b[1;32m    683\u001b[0m \u001b[38;5;66;03m# mess.\u001b[39;00m\n",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/urllib3/connectionpool.py:392\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    391\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 392\u001b[0m     \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mhttplib_request_kw\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    394\u001b[0m \u001b[38;5;66;03m# Reset the timeout for the recv() on the socket\u001b[39;00m\n",
      "File \u001b[0;32m/usr/lib/python3.11/http/client.py:1298\u001b[0m, in \u001b[0;36mHTTPConnection.request\u001b[0;34m(self, method, url, body, headers, encode_chunked)\u001b[0m\n\u001b[1;32m   1297\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Send a complete request to the server.\"\"\"\u001b[39;00m\n\u001b[0;32m-> 1298\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_send_request\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mencode_chunked\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/lib/python3.11/http/client.py:1344\u001b[0m, in \u001b[0;36mHTTPConnection._send_request\u001b[0;34m(self, method, url, body, headers, encode_chunked)\u001b[0m\n\u001b[1;32m   1343\u001b[0m     body \u001b[38;5;241m=\u001b[39m _encode(body, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mbody\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m-> 1344\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mendheaders\u001b[49m\u001b[43m(\u001b[49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mencode_chunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mencode_chunked\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/lib/python3.11/http/client.py:1293\u001b[0m, in \u001b[0;36mHTTPConnection.endheaders\u001b[0;34m(self, message_body, encode_chunked)\u001b[0m\n\u001b[1;32m   1292\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m CannotSendHeader()\n\u001b[0;32m-> 1293\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_send_output\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmessage_body\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mencode_chunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mencode_chunked\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/usr/lib/python3.11/http/client.py:1052\u001b[0m, in \u001b[0;36mHTTPConnection._send_output\u001b[0;34m(self, message_body, encode_chunked)\u001b[0m\n\u001b[1;32m   1051\u001b[0m \u001b[38;5;28;01mdel\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_buffer[:]\n\u001b[0;32m-> 1052\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmsg\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1054\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m message_body \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   1055\u001b[0m \n\u001b[1;32m   1056\u001b[0m     \u001b[38;5;66;03m# create a consistent interface to message_body\u001b[39;00m\n",
      "File \u001b[0;32m/usr/lib/python3.11/http/client.py:990\u001b[0m, in \u001b[0;36mHTTPConnection.send\u001b[0;34m(self, data)\u001b[0m\n\u001b[1;32m    989\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mauto_open:\n\u001b[0;32m--> 990\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    991\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/urllib3/connection.py:187\u001b[0m, in \u001b[0;36mHTTPConnection.connect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    186\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mconnect\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[0;32m--> 187\u001b[0m     conn \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_new_conn\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    188\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_prepare_conn(conn)\n",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/urllib3/connection.py:171\u001b[0m, in \u001b[0;36mHTTPConnection._new_conn\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    170\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m SocketError \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m--> 171\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m NewConnectionError(\n\u001b[1;32m    172\u001b[0m         \u001b[38;5;28mself\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mFailed to establish a new connection: \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m%\u001b[39m e\n\u001b[1;32m    173\u001b[0m     )\n\u001b[1;32m    175\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m conn\n",
      "\u001b[0;31mNewConnectionError\u001b[0m: <urllib3.connection.HTTPConnection object at 0x7f89c8fc2790>: Failed to establish a new connection: [Errno 111] Connection refused",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mMaxRetryError\u001b[0m                             Traceback (most recent call last)",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/requests/adapters.py:486\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    485\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 486\u001b[0m     resp \u001b[38;5;241m=\u001b[39m \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43murlopen\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    487\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    488\u001b[0m \u001b[43m        \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    489\u001b[0m \u001b[43m        \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    490\u001b[0m \u001b[43m        \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    491\u001b[0m \u001b[43m        \u001b[49m\u001b[43mredirect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    492\u001b[0m \u001b[43m        \u001b[49m\u001b[43massert_same_host\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    493\u001b[0m \u001b[43m        \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    494\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[1;32m    495\u001b[0m \u001b[43m        \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmax_retries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    496\u001b[0m \u001b[43m        \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    497\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    498\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    500\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ProtocolError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/urllib3/connectionpool.py:726\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    724\u001b[0m     e \u001b[38;5;241m=\u001b[39m ProtocolError(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mConnection aborted.\u001b[39m\u001b[38;5;124m\"\u001b[39m, e)\n\u001b[0;32m--> 726\u001b[0m retries \u001b[38;5;241m=\u001b[39m \u001b[43mretries\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mincrement\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    727\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43merror\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43me\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_pool\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m_stacktrace\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msys\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mexc_info\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m2\u001b[39;49m\u001b[43m]\u001b[49m\n\u001b[1;32m    728\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    729\u001b[0m retries\u001b[38;5;241m.\u001b[39msleep()\n",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/urllib3/util/retry.py:446\u001b[0m, in \u001b[0;36mRetry.increment\u001b[0;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[1;32m    445\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m new_retry\u001b[38;5;241m.\u001b[39mis_exhausted():\n\u001b[0;32m--> 446\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m MaxRetryError(_pool, url, error \u001b[38;5;129;01mor\u001b[39;00m ResponseError(cause))\n\u001b[1;32m    448\u001b[0m log\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mIncremented Retry for (url=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m): \u001b[39m\u001b[38;5;132;01m%r\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, url, new_retry)\n",
      "\u001b[0;31mMaxRetryError\u001b[0m: HTTPConnectionPool(host='127.0.0.1', port=5000): Max retries exceeded with url: /bluekeep_Dectect_Udec (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f89c8fc2790>: Failed to establish a new connection: [Errno 111] Connection refused'))",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mConnectionError\u001b[0m                           Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[3], line 17\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[38;5;66;03m# en este try se encapsula el valor devuelto por la api en una variable llamada response, que junto con la importacion requests se \u001b[39;00m\n\u001b[1;32m     15\u001b[0m \u001b[38;5;66;03m# llama el metodo post que es el encargado de determinar si la ip es vulnerable  o no\u001b[39;00m\n\u001b[1;32m     16\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m---> 17\u001b[0m     response \u001b[38;5;241m=\u001b[39m \u001b[43mrequests\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpost\u001b[49m\u001b[43m(\u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43mjson\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdata\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     18\u001b[0m     dat \u001b[38;5;241m=\u001b[39m response\u001b[38;5;241m.\u001b[39mjson()\n\u001b[1;32m     19\u001b[0m \u001b[38;5;66;03m# despues de convertir la respuesta en json, se trata la llave status, que trae una cadena de dos indices, y se llama al indice 1 que es el \u001b[39;00m\n\u001b[1;32m     20\u001b[0m \u001b[38;5;66;03m# encargado de traer la respuesta de vulnerabilidad, seguido a esto se ejecutan crea el contenedor, y el objeto que se enviará al metodo de \u001b[39;00m\n\u001b[1;32m     21\u001b[0m \u001b[38;5;66;03m# añadir items, el unico datos importante que no puede faltar es la respuesta en la variable cont, que tiene el contenedor ya creado,\u001b[39;00m\n\u001b[1;32m     22\u001b[0m \u001b[38;5;66;03m# esete comando se ejeutara si y solo si, la ip es vulnerable a bluekeep\u001b[39;00m\n",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/requests/api.py:115\u001b[0m, in \u001b[0;36mpost\u001b[0;34m(url, data, json, **kwargs)\u001b[0m\n\u001b[1;32m    103\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mpost\u001b[39m(url, data\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, json\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[1;32m    104\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Sends a POST request.\u001b[39;00m\n\u001b[1;32m    105\u001b[0m \n\u001b[1;32m    106\u001b[0m \u001b[38;5;124;03m    :param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    112\u001b[0m \u001b[38;5;124;03m    :rtype: requests.Response\u001b[39;00m\n\u001b[1;32m    113\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m--> 115\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mpost\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdata\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mjson\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mjson\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/requests/api.py:59\u001b[0m, in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     55\u001b[0m \u001b[38;5;66;03m# By using the 'with' statement we are sure the session is closed, thus we\u001b[39;00m\n\u001b[1;32m     56\u001b[0m \u001b[38;5;66;03m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001b[39;00m\n\u001b[1;32m     57\u001b[0m \u001b[38;5;66;03m# cases, and look like a memory leak in others.\u001b[39;00m\n\u001b[1;32m     58\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m sessions\u001b[38;5;241m.\u001b[39mSession() \u001b[38;5;28;01mas\u001b[39;00m session:\n\u001b[0;32m---> 59\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43msession\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/requests/sessions.py:589\u001b[0m, in \u001b[0;36mSession.request\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    584\u001b[0m send_kwargs \u001b[38;5;241m=\u001b[39m {\n\u001b[1;32m    585\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtimeout\u001b[39m\u001b[38;5;124m\"\u001b[39m: timeout,\n\u001b[1;32m    586\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m: allow_redirects,\n\u001b[1;32m    587\u001b[0m }\n\u001b[1;32m    588\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[0;32m--> 589\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mprep\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43msend_kwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    591\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/requests/sessions.py:703\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    700\u001b[0m start \u001b[38;5;241m=\u001b[39m preferred_clock()\n\u001b[1;32m    702\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[0;32m--> 703\u001b[0m r \u001b[38;5;241m=\u001b[39m \u001b[43madapter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msend\u001b[49m\u001b[43m(\u001b[49m\u001b[43mrequest\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    705\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n\u001b[1;32m    706\u001b[0m elapsed \u001b[38;5;241m=\u001b[39m preferred_clock() \u001b[38;5;241m-\u001b[39m start\n",
      "File \u001b[0;32m~/.local/lib/python3.11/site-packages/requests/adapters.py:519\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    515\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(e\u001b[38;5;241m.\u001b[39mreason, _SSLError):\n\u001b[1;32m    516\u001b[0m         \u001b[38;5;66;03m# This branch is for urllib3 v1.22 and later.\u001b[39;00m\n\u001b[1;32m    517\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m SSLError(e, request\u001b[38;5;241m=\u001b[39mrequest)\n\u001b[0;32m--> 519\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(e, request\u001b[38;5;241m=\u001b[39mrequest)\n\u001b[1;32m    521\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m ClosedPoolError \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[1;32m    522\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(e, request\u001b[38;5;241m=\u001b[39mrequest)\n",
      "\u001b[0;31mConnectionError\u001b[0m: HTTPConnectionPool(host='127.0.0.1', port=5000): Max retries exceeded with url: /bluekeep_Dectect_Udec (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f89c8fc2790>: Failed to establish a new connection: [Errno 111] Connection refused'))"
     ]
    }
   ],
   "source": [
    "#consumo de la api local para bluekeep\n",
    "#se consume el url de la api local, oero no la global local, para usar la global local en la api usada con python, \n",
    "#se le agrega el ejecutor de la varible inicializada con flask, donde esta el debug, se le agrega un host 0.0.0.0, para ejecutar\n",
    "#la api en un entorno local pero dentro de la red, usando la ip pública\n",
    "url = \"http://127.0.0.1:5000/bluekeep_Dectect_Udec\"\n",
    "# se pude la ip y e nombre dle contenedor, ay que tener en cuanta que la base de datos en este momento ya debe estar creada\n",
    "ipI = input(\"Digite la ip: \")\n",
    "nameContainer = input(\"Digite el nombre del contenedor: \")\n",
    "#se hace un pequeño objeto de datos que va a tener el ip ingresado y el puerto por defecto 3389\n",
    "data ={\n",
    "    \"ip\": ipI,\n",
    "    \"port\":3389\n",
    "}\n",
    "# en este try se encapsula el valor devuelto por la api en una variable llamada response, que junto con la importacion requests se \n",
    "# llama el metodo post que es el encargado de determinar si la ip es vulnerable  o no\n",
    "try:\n",
    "    response = requests.post(url,json=data)\n",
    "    dat = response.json()\n",
    "# despues de convertir la respuesta en json, se trata la llave status, que trae una cadena de dos indices, y se llama al indice 1 que es el \n",
    "# encargado de traer la respuesta de vulnerabilidad, seguido a esto se ejecutan crea el contenedor, y el objeto que se enviará al metodo de \n",
    "# añadir items, el unico datos importante que no puede faltar es la respuesta en la variable cont, que tiene el contenedor ya creado,\n",
    "# esete comando se ejeutara si y solo si, la ip es vulnerable a bluekeep\n",
    "    if dat['status'][1] == 'VULNERABLE':\n",
    "        cont = createNewContaniner(nameContainer)\n",
    "        datAS = {\n",
    "           'ip' : ipI,\n",
    "           'port' : data['port'],\n",
    "           'status' : dat['status'][1],\n",
    "           'id': nameContainer,\n",
    "           'name' : nameContainer,\n",
    "           'account' : 'bluekeep',\n",
    "           'container' : cont\n",
    "        }\n",
    "        addNewItem(datAS,cont,ipI)\n",
    "# en el caso que la ip no sea vulnerable se imprime ese mensaje\n",
    "    else:\n",
    "        print('La el dispositivo de esta ip está apagado, y se encuentra parchado para esta vulnerabilidad')\n",
    "        \n",
    "   #56 print(dat['status'][1])\n",
    "except requests.exceptions.HTTPError as http_err:\n",
    "    print(f'Error HTTP: {http_err}')\n",
    "    \n",
    "\n",
    "#ejemplos de vulnerabilidades 125.87.87.22 - 220.90.17.83"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "773bb4b1-81a8-4e52-8a8c-5f11b1683020",
   "metadata": {},
   "outputs": [],
   "source": [
    "#operaciones adicionales\n",
    "# informacion de la ip con io info\n",
    "#esta api fue obtenida de rapidApi, junto con la llave y el headers, se hace el llaamdo a la api dentro de un try catch, se opbtiene el vaor con \n",
    "# la variable resp, y se convierte a json con la variable response y la libreria json, luego se creal el objeto object co las llaves y la \n",
    "# informacion que consideré reelevante para guardar en mi base de datos, esta funcion retornará ese objeto si todo sale como se plantea\n",
    "def ipInfo(ip):\n",
    "    try:\n",
    "        \n",
    "        urlI = \"https://community-neutrino-ip-info.p.rapidapi.com/ip-info\"\n",
    "        payload = {\n",
    "    \t\"reverse-lookup\": \"checked\",\n",
    "    \t\"ip\": ip\n",
    "        }\n",
    "        headers = {\n",
    "    \t\"content-type\": \"application/x-www-form-urlencoded\",\n",
    "    \t\"X-RapidAPI-Key\": \"81eb58f566mshd228c280cb28728p1aa7c2jsnc1e57849a6ce\",\n",
    "    \t\"X-RapidAPI-Host\": \"community-neutrino-ip-info.p.rapidapi.com\"\n",
    "        }\n",
    "        resp = requests.post(urlI, data=payload, headers=headers)\n",
    "        response = resp.json()\n",
    "        object = {\n",
    "            'country': response['country'],\n",
    "            'city' : response['city'],\n",
    "            'region' : response['region'],\n",
    "            'timeZone' : response['timezone'],\n",
    "            'latitude' : response['latitude'],\n",
    "            'longitude' : response['longitude']\n",
    "        }\n",
    "        return object\n",
    "    except requests.exceptions.HTTPError as err:\n",
    "        print(f'Error HTTP: {err}')\n",
    "\n",
    "# uso de nmap\n",
    "# Se trató de usar nMap for python  pero datos a que no tenia los permisos respectivos con jupyter no podia ejecutar comandos \n",
    "# relacionados con tcp/ip, la funcion pide el ip y un argumento que está relacionado con la accion que ejecutará nMap, las operaciones\n",
    "# se ejecutan en un try catch para manejar los errores de consola, y que el proceso de la aplicacion no se vea afectado por algun error\n",
    "def nMap(ip,argument):\n",
    "    try: \n",
    "# aqui se pide la contraseña del super usuario con una libreria llamada get pass, es solo para que al momento de escribir no salgan letras\n",
    "# sino que salgan puntos, como cuando se dijita la contraseña en cualquier login bien hecho \n",
    "        password = getpass(\"Cuál es la contraseña Máquina ;): \")\n",
    "# este comando se divide en tres partes, y se ejecutara en la consola para llamar una accion de nmap\n",
    "# en la parte del echo, es la encargada de estar en espera a que la consola pida introducir algo, en este caso se asigana la contraseña ya \n",
    "# pedida, la segunda parte es donde se ejecuta el comando que llama a nmap con permisos root, se le asigan un argumento, y con el \n",
    "# --osscan-limit se limita a la busqueda de solo el sistema operativo, la tercera parte, con la palabra reservada greep, se filtra \n",
    "# la respuesta a un running(just guessing ) la cual trae la llave relacionada con el sistema operativo\n",
    "        comand = f'echo \"{password}\" | sudo -S nmap {argument} --osscan-limit {ip} | grep \"Running (JUST GUESSING)\" '\n",
    "# de aqui para abajo se obtiene el output con la libreria subprocess y un check_output, se activa el shell para permitir las interaccion con\n",
    "# consola, luego se convierte la respuesta a string, y se le hace un split en donde esta el :, y se retorna la aprte del split que corresponde\n",
    "# al indice 1\n",
    "        output = subprocess.check_output(comand, shell=True)\n",
    "        imp = output.decode('utf-8')\n",
    "        sO = imp.split(':')\n",
    "        return sO[1]\n",
    "# de no poder ejecutarse correctamente, simplemente se retorna el mensaje de No encontrado para guardar en la abse de datos \n",
    "    except subprocess.CalledProcessError as e:\n",
    "        return 'No encontrado'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "79e705e6-8588-4b5f-8ae3-8518560ceab8",
   "metadata": {},
   "outputs": [],
   "source": [
    "#creacion del contenedor para las pruebas y todo lo que tiene que ver con azure\n",
    "#definicion de permisos y conexion\n",
    "# aqui se gregan las credenciales y el nombre de la base de datos a crearse o a llamer\n",
    "settings = {\n",
    "    'host': os.environ.get('ACCOUNT_HOST', 'https://nicolasdb.documents.azure.com:443/'),\n",
    "    'master_key': os.environ.get('ACCOUNT_KEY', 'fvxrIjHEajqpiTJq5HCSJLYiyBJHoAELiaNO23knR5LZ5rk4iEgE8CNCO37lGTg0kjhdy5m1KE9RACDbA2JsSQ=='),\n",
    "    'database_id': os.environ.get('COSMOS_DATABASE', 'nicolasdb'),\n",
    "}\n",
    "\n",
    "#Asignación de valores a variables internas\n",
    "HOST = settings['host']\n",
    "MASTER_KEY = settings['master_key']\n",
    "DATABASE_ID = settings['database_id']\n",
    "try:\n",
    "    # aqui se creal el cliente que es como la puerta de entrada a la base de datos \n",
    "    client = cosmos_client.CosmosClient(HOST, {'masterKey': MASTER_KEY}, user_agent=\"CosmosDBPythonQuickstart\", user_agent_overwrite=True)\n",
    "    # con esa misma puerta de entrada se hace un llamado a la base de datos y se guarad en esa variable db\n",
    "    db = client.get_database_client('nicolasdb')\n",
    "except exceptions.CosmosHttpResponseError as e:\n",
    "        print('\\nexpecion controlada: '.format(e.message))\n",
    "\n",
    "def createNewContaniner(name):\n",
    "# en esta parte se crea el contenedor con la base de datos ya obtenida, pero si no se ejecuta el comando de crear, quiere decir\n",
    "# que el contenedor ya está creado o paso algo mas, pero nuestro cath, lo que haces procesar que el contenedor ya está creado y de ser asi,\n",
    "# simplemenet lo llama y lo retorna, y en cualquiera de los casos siempre se va a retornar el contenedos\n",
    "        try:\n",
    "            container = db.create_container(id=name, partition_key=PartitionKey(path='/partitionKey'))\n",
    "            print('Container with id \\'{0}\\' created'.format(name))\n",
    "\n",
    "        except exceptions.CosmosResourceExistsError:\n",
    "            container = db.get_container_client(name)\n",
    "            print('Container with id \\'{0}\\' was found'.format(name))\n",
    "        finally:\n",
    "            return container\n",
    "\n",
    "# esta es la funcion encargada de crear el item, esta pide el contenedor, un objeto y la ip\n",
    "def addNewItem(object,container,ip):\n",
    "        try:\n",
    "          #   print(\"Al añadir \",ip)\n",
    "# primero se pone en ejecucion un while que va a correr infinitamente hasta que se cumpla la lavildacion que estpa entro de este,\n",
    "# primero se pide el nombre del item luego se crea un objeto llamdo info, con el contenedor, el nombre y el account es la subdivision,\n",
    "# luego de esto se crean dos variables que va a retornar una funcion, la cual es read_item, la cual se encarga de mirar si el item existe,\n",
    "# esta retorna uno, si existe retorna true y el while vuelve a ejecutar la funcion desde el principio, de retornar false, esto queire decir, que \n",
    "# el item no existe, entonces se entra al if porque cumple con la validación, la cual ejecuta la funcion de nmap ya explicada, y se le \n",
    "# envia el atributo de la ip, y el -O que es el encargado de revisar los sitemas operativos en nmap, luego de esto se crea un objeto,\n",
    "# que es ya el que se va a designar para crear el item en la base de datos, guaradando la data de el tiempo en que se ejecutó, se guarda el \n",
    "# sistema oprativo, que es repuesta de la funcion de nmap, y una llave ipinfo que llama a la funcion de ip info, esto con el fin de,\n",
    "# separar todo lo que retorna nustra funcion en una sola llave \n",
    "             while True:\n",
    "                 name = input('Digite el nombre del item: ')\n",
    "                 inf = {\n",
    "                     'container' : container,\n",
    "                     'id' : name,\n",
    "                     'account' : 'bluekeep'\n",
    "                 }\n",
    "                 bol,result = read_item(inf)\n",
    "                 if bol == False:\n",
    "                         nm = nMap(ip,'-O')\n",
    "                         dataS = {'id' : name,\n",
    "                        'partitionKey' : \"bluekeep\",\n",
    "                        'date' : datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S'),\n",
    "                        'port' : object['port'],\n",
    "                        'systemOp' : nm,\n",
    "                        'status' :object['status'],\n",
    "                        'ipInfo' : ipInfo(object['ip'])\n",
    "                                 }\n",
    "                         print(nm)\n",
    "                         container.create_item(body=dataS)\n",
    "                         break\n",
    "            \n",
    "             \n",
    "             print(f\"item {name} fue creado con exito\")\n",
    "        except exceptions.CosmosResourceExistsError as err:\n",
    "            print('Container with id \\'{0}\\' was found'.format(err))\n",
    "    \n",
    "#el objecto debe dar el contenedor <3\n",
    "# esta funcion se enarga de eliminar items, solo con le nombre o id, el contenedor y el account\n",
    "def deleteItem(object):\n",
    "    try:\n",
    "        object['container'].delete_item(item=object['id'], partition_key=object['account'])\n",
    "        print('item eliminado')\n",
    "    except exceptions.CosmosHttpResponseError as e:\n",
    "        error = str(e)\n",
    "        print(\"Excepción controlada: \"+error)\n",
    "\n",
    "#leer item\n",
    "# esta es la funcion que lee la base de datos en busqueda de un item en especifico\n",
    "def read_item(object):\n",
    "# todo lo que se ejecuta dentro del try no es más que un llamado o busqueda de un item en específico, en casod e encontrarlo,\n",
    "# se retorna un boleano verdadero y una respuesta que es lo que trae ese item, en caso de que se llegase a usar en con fines de saber que hay \n",
    "# en el item, en caso de no encontrarse se entra en la excepcion de notfound que trae cosmos, y se retorna falso seguido de un mensaje,\n",
    "# de no existencia del item\n",
    "    try:\n",
    "        print('\\nReading Item by Id\\n')\n",
    "        response = object['container'].read_item(item=object['id'], partition_key=object['account'])\n",
    "        ##print('Partition : {0}'.format(response))\n",
    "        return True,response\n",
    "    except exceptions.CosmosResourceNotFoundError:\n",
    "        return False,'El item no fue encontrado'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f6e99d51-72c0-4e68-9cab-a8b54911a268",
   "metadata": {},
   "outputs": [],
   "source": [
    "#cont = createNewContaniner('pruebaBluekeep1')\n",
    "#obj = {\n",
    "#    'container' : cont ,\n",
    "#    'id' : 'pruebaitem',\n",
    "#    'account' : 'bluekeep'\n",
    "#}\n",
    "#bol,response = read_item(obj)\n",
    "#print(response)\n",
    "\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06fbf308-6ed8-4d05-a685-be87f7227743",
   "metadata": {},
   "outputs": [],
   "source": [
    "#operaciones de prueba, recordar simbre usar primero el crador de contenedores para recuperar el contenedor creado\n",
    "#cont = createNewContaniner(\"prueba2MAR\")\n",
    "#object = {\n",
    "#    'id': 'pruebaitem1',\n",
    "#    'name' : 'pruebaitem1',\n",
    "#    'account' : 'bluekeep',\n",
    "#    'container' : cont\n",
    "#}\n",
    "#deleteItem(object)\n",
    "#addNewItem(\"1\",cont)\n",
    "#read_item(object)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "95c1741a-719f-4b1c-8c4a-b691ebcfded0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#ipInfo('121.33.10.8')\n",
    "#nameContainer = input(\"Digite el nombre del contenedor: \")\n",
    "#dn = nMap('118.68.123.237','-O')\n",
    "#print(dn)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd92f5b6-c30f-43e8-9f1d-986c18317c44",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28ee20f4-c259-43d8-9f0c-b521e2559af1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c340ad23-dccb-463d-88f2-c7c2bed3f9c1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
